### PR TITLE
feat: loader skeletons

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### <samp>🤖 Generated by Copilot at ddc93ba</samp>

This pull request improves the loading states of various components in the `journeys-admin` app, using the MUI `Skeleton` component to show placeholders while the journey data is loading. It also adds or updates test cases and stories for these components, and fixes some minor code issues. The affected components are `SocialImage`, `TemplateCardPreview`, `TemplateView`, and `TemplateViewHeader`.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/34356579/todos/6644505406)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] it should show skeletons while loading

# Walkthrough

### <samp>🤖 Generated by Copilot at ddc93ba</samp>

*  Add skeleton placeholders for the loading state of various components that render journey data ([link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-b07106c3731edc1214b7541edfafab564079443c1b926e85a95b2be84780fb23R61-R69), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-d49917dceed339d17f11a2e279bd1fe087f175c393633b004fc8dbaeb06d8835R65-R71), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-2f472b03d2f1e26b611957f1864d7870c6fcd7adfe64f39533e543a2485d0b6dR2), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-2f472b03d2f1e26b611957f1864d7870c6fcd7adfe64f39533e543a2485d0b6dL52-R61), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-d0256bffa008bf02d202eeadafe9f56a2d341b0ea3a6ab7794f99c576ffe4314R278-R296), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-18cec4c017b3d56f23d82e6016eadd161d82c45d46b1cba614bf90c0ba722497R104), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-b01abe788f5622371478021f7d3e86b2ef600cb38caff63581cbbbc872820294R52-R64), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-bac56af7a9a697c296e47ed332cfee75f0a9892398f94ea4e182a688d6475781L25-R39), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-89a406c95c26ccbfb9155819b7a64eb0357873b6f735f62df2036871c786fceaR4), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-89a406c95c26ccbfb9155819b7a64eb0357873b6f735f62df2036871c786fceaL118-R163), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-069b2c2fdb60e59618cfde000ce54408a7c5018bd7f92003b1e16e6d5822327dR165-R177), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-25131dd9028eed90b85e738c192f969471e3c14909b07851f3ecb773cc3026fcR441-R449), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150R3), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150L54-R63), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-a7f97c29bef72095eee7e677a39f5819efb129784d3bf53cd2171569d540b6ecR179-R195), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-d5a4410a9da4454f3b1d035291a6674078dbeb1d48708d12070d720ff8de326eR81-R89), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-ee24d1185380438fa67baa8ace69d9cdfb3ae3e712ce48150234939ea8ea096bR2), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-ee24d1185380438fa67baa8ace69d9cdfb3ae3e712ce48150234939ea8ea096bL32-R55), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-ee24d1185380438fa67baa8ace69d9cdfb3ae3e712ce48150234939ea8ea096bL60-R105), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-ee24d1185380438fa67baa8ace69d9cdfb3ae3e712ce48150234939ea8ea096bL95-R126))
* Remove unnecessary optional chaining operator from the `journeys` prop in the `TemplateSection` component ([link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-0016fcb7bad744aecc9ef2abfad388fd2d490ec024dc331c2800e4f4bd2069b0L119-R119))
* Rename the imported `JourneyFields` type to `Journey` in the stories for the `TemplatePreviewTabs` and `TemplateView` components ([link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-080d8aa134f161e27b912696140af965864ce69584482aa845bfcec696bfa65aL2-R6), [link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-25131dd9028eed90b85e738c192f969471e3c14909b07851f3ecb773cc3026fcL13-R16))
* Remove an empty line in the `TemplateCardPreview` component ([link](https://github.com/JesusFilm/core/pull/1961/files?diff=unified&w=0#diff-89a406c95c26ccbfb9155819b7a64eb0357873b6f735f62df2036871c786fceaR47))

### CODE REVIEW

File: apps/journeys-admin/src/components/JourneyView/SocialImage/SocialImage.tsx [hunk 1]
```diff
@@ -49,7 +50,15 @@ export function SocialImage({
             backgroundColor: 'background.default'
           }}
         >
-          {journey != null ? <GridEmptyIcon fontSize="large" /> : <></>}
+          {journey != null ? (
+            <GridEmptyIcon fontSize="large" />
+          ) : (
+            <Skeleton
+              data-testid="socialImageSkeleton"
```
[the remark this answers is not in the document]
Comment (csiyang) [reply apps/journeys-admin/src/components/JourneyView/SocialImage/SocialImage.tsx:57]: Capitalize the other `data-testid`s you added too, make sure all of them are consistent

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.tsx [hunk 2]
```diff
@@ -29,23 +30,29 @@ export function TemplateViewHeader({
...
+        {journey?.createdAt != null ? (
+          intlFormat(parseISO(journey?.createdAt), {
             month: 'long',
             year: 'numeric'
-          })}
-        </Typography>
-      )}
+          })
+        ) : (
+          <Skeleton
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.tsx:49]: move skeleton outside of typog

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplateCardPreview/TemplateCardPreview.tsx [hunk 3]
```diff
@@ -115,24 +117,50 @@ export function TemplateCardPreview({
...
-            width: smUp ? '240px' : '177px',
-            height: smUp ? '380px' : '280px',
-            zIndex: 2
-          }}
-        >
-          <TemplateCardPreviewItem step={step} smUp={smUp} />
-        </SwiperSlide>
-      ))}
+      {steps != null
+        ? steps?.map((step) => {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplateCardPreview/TemplateCardPreview.tsx:126]:
```suggestion
        ? steps.map((step) => {
```

File: apps/journeys-admin/src/components/TemplateView/TemplateView.tsx [hunk 4]
```diff
@@ -51,7 +52,15 @@ export function TemplateView({ authUser }: TemplateViewProps): ReactElement {
...
           sx={{ display: { xs: 'block', sm: 'none' } }}
         >
-          {journey?.description}
+          {journey?.description != null ? (
+            journey.description
+          ) : (
+            <>
+              <Skeleton data-testid="template-view-skeleton" width="100%" />
+              <Skeleton data-testid="template-view-skeleton" width="100%" />
+              <Skeleton data-testid="template-view-skeleton" width="100%" />
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplateView.tsx:61]: use array

Comment (csiyang) [reply apps/journeys-admin/src/components/TemplateView/TemplateView.tsx:61]: Also do this every other place where your indivudually rendering skeletons

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.tsx (hunk 2 above)
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.tsx:49]: interesting.... is there a reason we don't use it this way: ?
https://mui.com/material-ui/react-skeleton/#inferring-dimensions

File: apps/journeys-admin/src/components/JourneyView/SocialImage/SocialImage.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (csiyang) [reply apps/journeys-admin/src/components/JourneyView/SocialImage/SocialImage.tsx:57]: You haven't updated this yet

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplateCardPreview/TemplateCardPreview.tsx [hunk 5]
```diff
@@ -121,18 +123,43 @@ export function TemplateCardPreview({
...
+                style={{
+                  width: 'fit-content',
+                  zIndex: 2
+                }}
+              >
+                <TemplateCardPreviewItem step={step} />
+              </SwiperSlide>
+            )
+          })
+        : [0, 1, 2, 3, 4, 5, 6].map((i) => {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplateCardPreview/TemplateCardPreview.tsx:141]: Typically `i` is used for index, we should use something else

Comment (csiyang) [reply apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplateCardPreview/TemplateCardPreview.tsx:141]: also make this change in the other places you have just used `i`

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.tsx [hunk 6]
```diff
@@ -29,23 +30,29 @@ export function TemplateViewHeader({
...
-          sx={{
-            color: 'secondary.light',
-            display: { xs: 'block', sm: 'none' },
-            pb: 6
-          }}
-          noWrap
-        >
-          {intlFormat(parseISO(journey?.createdAt), {
+      <Typography
+        data-testId="CreatedAtTemplatePreviewPage"
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.tsx:34]: Why so many test ids on this component? If you really need one,  a `data-testid = TemplateViewHeader` on the top level stack should be sufficient (keep the skeleton ones). You shouldn't need the test-ids here to test this component, test for the actual values.

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx [hunk 7]
```diff
@@ -155,8 +155,8 @@ describe('TemplateViewHeader', () => {
     expect(getAllByRole('link', { name: 'Preview' })[0]).toBeInTheDocument()
   })
 
-  it('should show featured date if journey is featured', () => {
-    const { getAllByTestId } = render(
+  it('should show created at date', () => {
+    const { getAllByText } = render(
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx:159]: Use getBy for single elements, or check for length if there are more than 1
```suggestion
    const { getByText } = render(
```

**Review by csiyang — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx (hunk 7 above)
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateView/TemplateViewHeader/TemplateViewHeader.spec.tsx:159]: this is one of those elements that have xs: hidden sm: block so they still show up twice in the dom. Maybe a better way I can assert this is     `expect(getAllByText('November 2021')).toHaveLength(2)`

File: apps/journeys-admin/src/components/TemplateView/TemplateView.tsx [hunk 8]
```diff
@@ -52,15 +53,27 @@ export function TemplateView({ authUser }: TemplateViewProps): ReactElement {
...
         >
-          {journey?.description}
+          {journey?.description != null ? (
+            journey.description
+          ) : (
+            <>
+              {[0, 1, 2].map((value) => (
+                <Skeleton
+                  key={value}
+                  data-testid="TemplateViewSkeleton"
```
Comment (csiyang) [inline apps/journeys-admin/src/components/TemplateView/TemplateView.tsx:63]:
```suggestion
                  data-testid="TemplateViewDescriptionSkeleton"
```

**Review by csiyang — CHANGES_REQUESTED** [review]
Minor name change

**Review by csiyang — APPROVED** [review]

**Pull request merged**